daprdocs/content/en/reference/api/distributed_lock_api.md

```diff
@@ -0,0 +1,150 @@
+---
+type: docs
+title: "Distributed Lock API reference"
+linkTitle: "Distributed Lock API"
+description: "Detailed documentation on the distributed lock API"
+weight: 660
+---
+
+## Lock
+
+This endpoint lets you acquire a lock.
+
+### HTTP Request
+
+```
+POST http://localhost:<daprPort>/v1.0-alpha1/lock/<storename>
+```
+
+#### URL Parameters
+
+Parameter | Description
+--------- | -----------
+`daprPort` | The Dapr port
+`storename` | The `metadata.name` field component file. Refer to the [component schema] ({{< ref component-schema.md>}})
+
+#### Query Parameters
+
+None
+
+### HTTP Response codes
+
+Code | Description
+---- | -----------
+200  | Request successful
+204  | Empty Response
+400  | Malformed request
+500  | Request failed
+
+### HTTP Request Body
+
+The lock endpoint receives the following JSON payload:
+
+```json
+{
+    "resourceId": "",
```

Comment:
It would be good to have a short explanation for each of these fields, especially what resourceId and lockOwner are used for. Need to tie this into the docs for each lock having a name which I presume is the lockOwner. Can the resource Id be anything? What am I expected to put here?

Reply:
Added